hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -1488,6 +1488,61 @@ def hpat_pandas_series_unique_impl(self):
     return hpat_pandas_series_unique_impl
 
 
+@overload_method(SeriesType, 'cumsum')
+def hpat_pandas_series_cumsum(self, axis=None, skipna=True, *args):
+    """
+    Pandas Series method :meth:`pandas.Series.cumsum` implementation.
+
+    .. only:: developer
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_cumsum
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_cumsum_unboxing
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_cumsum_full
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_cumsum_str
+       Test: python -m hpat.runtests hpat.tests.test_series.TestSeries.test_series_cumsum_unsupported_axis
+
+    Parameters
+    ----------
+    self: :obj:`pandas.Series`
+        input series
+    axis: :obj:`int`, :obj:`str`
+        Axis along which the operation acts
+        0/None - row-wise operation
+        1      - column-wise operation
+        *unsupported*
+    skipna: :obj:`bool`
+        exclude NA/null values
+    *args:
+        *unsupported*
+
+    Returns
+    -------
+    :obj:`pandas.Series`
+         returns :obj:`pandas.Series` object
+    """
+
+    _func_name = 'Method cumsum().'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError('{} The object must be a pandas.series. Given self: {}'.format(_func_name, self))
+
+    if not isinstance(self.dtype, types.Number):
+        raise TypingError('{} The object must be a number. Given self.dtype: {}'.format(_func_name, self.dtype))
```

Comment:
see https://github.com/IntelPython/hpat/pull/216#discussion_r335117490
also, message `Given self.dtype` means nothing for user. Please keep messages understandable as much as possible